lib/gettext/po/translations.ex

```diff
@@ -28,9 +30,38 @@ defmodule Gettext.PO.Translations do
 
   """
   @spec autogenerated?(PO.translation()) :: boolean
-  def autogenerated?(%module{flags: flags} = _translation) when is_translation(module) do
-    # TODO: Remove elixir-format flag to detect autogenerated translations in future releases
-    MapSet.member?(flags, "ex-autogen") or MapSet.member?(flags, "elixir-format")
+  def autogenerated?(%module{msgid: msgid, flags: flags} = _translation)
+      when is_translation(module) do
+    cond do
+      MapSet.member?(flags, "elixir-autogen") ->
+        true
+
+      MapSet.member?(flags, "ex-autogen") ->
+        Logger.warn("""
+        The `ex-autogen` flag is deprecated.
+        Please run gettext extract & merge to replace it with the newest flag.
+
+        Translation:
+        #{msgid}
+        """)
+
+        true
+
+      MapSet.member?(flags, "elixir-format") ->
+        Logger.warn("""
+        The `elixir-format` flag is deprecated for detecting autogenerated translations.
+        In a future version, translations without the `elixir-autogen` flag will not be purged when no longer used.
+        Please run gettext extract & merge to replace it with the newest flag.
+
+        Translation:
+        #{msgid}
+        """)
+
+        true
+
+      true ->
+        false
+    end
```

Comment:
I would not worry about this for now. Maybe we can introduce it 2 or 3 versions down the road because they should have upgraded by then. We probably don't even need to check for ex-autogen because most people will still have elixir-format and the one person who might not is the one drafting this PR. :D

```suggestion
    MapSet.member?(flags, "elixir-autogen") or MapSet.member?(flags, "elixir-format")
```

Reply:
@josevalim I reverted to the simple condition without logging but included `ex-autogen`. Is that ok?